src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -675,31 +680,21 @@ private byte[] engineDecryptFinal(byte[] input, final int offset, final int leng
         input = EMPTY_ARRAY;
       }
 
-      // If we already have processed some amount of ciphertext then we must coallesce it prior to
-      // decryption.
-      // Otherwise, we can just use the provided input.
+      checkOutputBuffer(inputLen, output, outputOffset, true);
+      checkArrayLimits(input, inputOffset, inputLen);
+
       final byte[] workingInputArray;
       final int workingInputOffset;
       final int workingInputLength;
-      if (decryptInputBuf.size() > 0) {
-        // We've already processed data so need to merge this in
-        engineUpdate(input, offset, length); // Decrypt mode never generates output for updates
-        workingInputArray = decryptInputBuf.getDataBuffer();
-        workingInputLength = decryptInputBuf.size();
-        workingInputOffset = 0;
-      } else {
```

Comment:
This is a small step backwards in favor of making things more uniform. Previously, we were able to get away with using the input buffer directly when the client calls doFinal using the byte[] interfaces and without calling doUpdate prior. With this change, internally buffering the input on decryption operations is now compulsory. If we have telemetry to suggest that this particular call pattern is common enough, we can try to add some logic to accommodate it. FWIW, Sun JSSE's TLS implementation calls into our library via the ByteBuffer interface where this has no impact.